src/permutation/permutation.rs

```diff
@@ -565,174 +507,229 @@ impl Permutation {
                 prev.6 *= current_component.6;
                 prev.7 *= current_component.7;
 
-                prev
-            })
-            .collect();
+                prev
+            })
+            .collect();
+
+        // Right now we basically have 6 acumulators of the form:
+        // A1 = [a1, a1 * a2, a1*a2*a3,...]
+        // A2 = [b1, b1 * b2, b1*b2*b3,...]
+        // A3 = [c1, c1 * c2, c1*c2*c3,...]
+        // ... and so on
+        // We want:
+        // [a1*b1*c1, a1 * a2 *b1 * b2 * c1 * c2,...]
+        // Parallisable
+        let mut z: Vec<_> = product_acumulated_components
+            .iter()
+            .map(move |current_component| {
+                let mut prev = BlsScalar::one();
+                prev *= current_component.0;
+                prev *= current_component.1;
+                prev *= current_component.2;
+                prev *= current_component.3;
+                prev *= current_component.4;
+                prev *= current_component.5;
+                prev *= current_component.6;
+                prev *= current_component.7;
+
+                prev
```

Comment:
This is a clear example, looks like `current_component` is a tuple of elements of the same type; if it was an array you could just loop it or call `iter().product()` or `fold`.

Reply:
Will open an issue for this one since will take me a bit more time to figure out propperly and optimally

Reply:
https://github.com/dusk-network/plonk/issues/562 Filled